cub/cub/util_type.cuh

```diff
@@ -46,24 +46,24 @@
 #include <cuda/__cccl_config> // _CCCL_CUDACC_VER
 #include <cuda/std/limits>
 
-#if !_NVHPC_CUDA
+#if !defined(_CCCL_CUDA_COMPILER_NVHPC) && !defined(CUB_DISABLE_FP16_SUPPORT)
 #  include <cuda_fp16.h>
-#endif
+#endif // !_CCCL_CUDA_COMPILER_NVHPC && !CUB_DISABLE_FP16_SUPPORT
+
+#if !defined(CUB_DISABLE_BF16_SUPPORT) && defined(CUB_DISABLE_FP16_SUPPORT)
+#  define CUB_DISABLE_BF16_SUPPORT
+#endif // !CUB_DISABLE_BF16_SUPPORT && CUB_DISABLE_FP16_SUPPORT
```

Comment:
suggestion: it's not exactly trivial that bf16 includes fp16, and if fp16 is disabled we also have to disable bf16. Could you please add a comment? Also, we might want to document this for users:
```cpp
#ifdef DOXYGEN_SHOULD_SKIP_THIS

/**
 * @def CCCL_DISABLE_FP16_SUPPORT
 *
 * Has to disable BF16 support as well, etc.
 */
#  define CCCL_DISABLE_FP16_SUPPORT

#endif
```